File: backend/src/main/scala/admin/api/ResolveReviewAppealApi.scala
package admin.api

import domain.shared.given

import cats.effect.IO
import domain.review.ResolveReviewAppealRequest
import domain.shared.{DeliveryAppState, ReviewAppealId, UserRole}
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityCodec.*
import review.app.*
import shared.api.routing.*

val resolveReviewAppealApi: FixedMethodApi1[ReviewAppealId, DeliveryAppState] =
  jsonPostApi1[ReviewAppealId, ResolveReviewAppealRequest, DeliveryAppState](
    List(routeSegment("api"), routeSegment("delivery"), routeSegment("review-appeals")),
    List(routeSegment("review")),
  )

File: frontend/src/apis/admin/ResolveReviewAppealApi.ts
import type { DeliveryAppState, ResolveReviewAppealRequest, ReviewAppealId } from '@/objects/core/SharedObjects'
import { postNormalizedDeliveryState } from '@/system/api/DeliveryStateClient'
import { buildApiPath1, defineJsonPostApi1, routeSegment } from '@/system/api/TypedApiDefinitions'

export const resolveReviewAppealApiDefinition = defineJsonPostApi1<ReviewAppealId, ResolveReviewAppealRequest, DeliveryAppState>(
    [routeSegment('api'), routeSegment('delivery'), routeSegment('review-appeals')],
    [routeSegment('review')],
  )

export function resolveReviewAppeal(
  appealId: ReviewAppealId,
  payload: ResolveReviewAppealRequest,
) {
  return postNormalizedDeliveryState(buildApiPath1(resolveReviewAppealApiDefinition, appealId), payload)
}
